### ISSUE TITLE
TracerV-based Targets using BaseVitisConfig

### ISSUE THREAD

**Issue by davidbiancolin** [firesim/firesim#1305]
### Background Work

- [X] Yes, I searched the [mailing list](https://groups.google.com/forum/#!forum/firesim)
- [X] Yes, I searched [prior issues](https://github.com/firesim/firesim/issues)
- [X] Yes, I searched the [documentation](https://docs.fires.im/en/stable/)

### FireSim Version and Hash

main (11/17/2022)
SHA: 4cd74f89 

### OS Setup

firesim1, azure instances with vitis 2022.1

### Current Behavior

FPGA deployements and metasims deadlock at cycle 2.

### Expected Behavior

They should simulate.

### Other Information

I dug into this a bit and it seems like the token channels for triggerCredits and triggerDebits are not being driven. This suggests to me that something about the assignment to these channels from the bridges has broken, likely as a side effect of changing some channel handling logic in SimWrapper.

**Comment by davidbiancolin** [comment]
@nandor FYI

**Comment by PKUZHOU** [comment]
Have you found a solution? I am also faced with this problem...

**Comment by abejgonzalez** [comment]
I can confirm this is also happening on a non-TracerV target: https://github.com/firesim/firesim/actions/runs/3755014646/jobs/6379727415

**Comment by davidbiancolin** [comment]
Are you certain it's non-tracerv? If so can you provide the cause?

**Comment by PKUZHOU** [comment]
By using metasim, I have found that the TracerVBridge has broken.  The TracerV's channel always has out_ready = 0 so that the channel gets stuck at cycle 2:
<img width="1124" alt="image" src="https://user-images.githubusercontent.com/25677298/209176836-1df29662-fc80-48c2-b3a6-420b89736e32.png">
In the generated verilog file, I can see that this is because the io_out_ready signal is constantly set to zero. 
<img width="1211" alt="image" src="https://user-images.githubusercontent.com/25677298/209176504-7d5f91aa-e6b4-4bc0-964f-b0e84dbb67ee.png">

I can also find that some ports of TracerVBridge are missing:
<img width="602" alt="image" src="https://user-images.githubusercontent.com/25677298/209175436-96251221-e98b-46c3-b4ff-35ddd037f124.png">

Here is a functional version derived from https://github.com/EECS-NTNU/firesim/tree/1.15.1_u250

<img width="1034" alt="image" src="https://user-images.githubusercontent.com/25677298/209175972-cc1034b7-b1ad-424f-b1cf-bdf970dcbf41.png">

Do you have any ideas about this? It looks like TracerVBridge is not correctly connected into the simulator during the compiling phase.

**Comment by davidbiancolin** [comment]
Yes, this is consistent with the initial bug report. @PKUZHOU what version of FireSim are you using?

**Comment by PKUZHOU** [comment]
Hi @davidbiancolin, I have tested both 1.15.1 and  the main branch, they have the same problem. By the way, I have also tested the micro2022_tutorial branch but it also got stuck at Cycle 2

**Comment by davidbiancolin** [comment]
Ok thanks.

**Comment by davidbiancolin** [comment]
Ok thanks.

**Comment by PKUZHOU** [comment]
By removing WithTracerVBridge from the config, firesim works well on my U200 FPGA. Branch 1.15.1

**Comment by davidbiancolin** [comment]
Recent changes (went in yesterday) i suspect resolved this indirectly. 

The route cause if i had to guess was that because there was no PCIM, GG/FIRRTL optimized away all of the tracerV logic and tied off the channels.

**Comment by PKUZHOU** [comment]
Hi, David

May I ask if you have solved this issue?
